### ISSUE TITLE
Vagrant cannot update plugin dependencies (semantic versioning is not possible)

### ISSUE THREAD

**Issue by maxlinc** [hashicorp/vagrant#2499]
I tried to make vagrant-rackspace use the recommended two-digit pessimistic constraint for versioning in https://github.com/mitchellh/vagrant-rackspace/pull/49.  Currently, this should have caused it to install fog 1.18.0 instead of 1.15.0.

In my testing, Vagrant did not behave that way:
- Installing from a gem (vagrant plugin install pkg/vagrant-rackspace-0.1.5.gem) worked as expected.  Fog 1.18.0 was installed.
- Installing from a local gem server (`vagrant plugin install --plugin-source http://localhost:8808 vagrant-rackspace`) did not work.  Fog 1.15.0 was installed.

I don't think it is a bug in my gem server, because bundle resolves the correct version with the source set to http://localhost:8808.

The RubyGem guidelines are definitely in favor of semantic versioning.  The [guide](http://guides.rubygems.org/patterns/#semantic_versioning).  I noticed there have been major [RubyGems changes](https://github.com/rubygems/rubygems/blob/master/History.txt) since the 1.8.23 version that currently ships with Vagrant.  That includes:
- RubyGems uses a new dependency resolver for gem installation which works
  similar to the bundler resolver.  The new resolver can resolve conflicts the
  previous resolver could not and offers improved diagnostics when conflicts
  are discovered.

In summary: `vagrant plugin update <plugin_name>` updates the plugin, but does not seem to update plugin dependencies to the newest compatible version.
